bin/garden-feat

```diff
@@ -0,0 +1,235 @@
+    try:
+        with open(fname) as f:
+            content = yaml.load(f, Loader=yaml.FullLoader)
+            return content
+    except FileNotFoundError:
+        # Avoid creating an unexpected artifact with missing
+        # features
+        print(f'garden-feat: error: Could not find feature: {fname}')
+        sys.exit(1)
+
+
+def get_all_features(feature_dir, features, features_ignore, features_matrix):
+    """ Get all features including their include and exclude features """
+    # Convert feature list to Python list
+    features_include = conv_features_list(features)
+    features_exclude = conv_features_list(features_ignore)
+    features_matrix = {}
+
+    # Run recursion to gfetch all features and their include features
+    get_include_features(feature_dir, features_matrix, features_include, features_exclude)
+    return features_matrix, features_exclude
+
+
+def get_include_features(feature_dir, features_matrix, features_include, features_exclude):
+    """ Get features included by other features """
+    if len(features_include) > 0:
+        # Loop for feature in "features_include"
+        for i in features_include:
+
+            # Proceed with given feature (yaml) file
+            fname = f"{feature_dir}/{i}/info.yaml"
+            yaml_content =  get_yaml_content(fname)
+            feature_features = yaml_content.get('features', None)
+            # Write content for each loaded feature in dict
+            # to avoid re-reading the file content
+            features_matrix[i] = yaml_content
+            # Remove proceeded feature from "features_include" list
+            features_include.remove(i)
+
+            # Proceed with feature include from feature by iterator
+            if feature_features is not None:
+                _features_include = feature_features.get('include', None)
+            else:
+                _features_include = None
+
+            # Do not add features by include if main features is excluded
+            if not i in features_exclude:
+                # Add all included features if present or given by config
+                if _features_include:
+                    for i in _features_include:
+                        if not i in features_include:
+                            features_include.append(i)
+
+            # Proceed with feature exclude from feature by iterator
+            if feature_features is not None:
+                _features_exclude = feature_features.get('exclude', None)
+            else:
+                _features_exclude = None
+            # Add all included features if present or given by config
+            if _features_exclude:
+                for i in _features_exclude:
+                    if not i in features_exclude:
+                        features_exclude.append(i)
+
+        # Run recursion to fetch includes from includes
+        get_include_features(feature_dir, features_matrix, features_include, features_exclude)
```

Comment:
I wouldn't call this function like this but move it within the for loop. This should help omitting the list removals of feature_include etc.

Reply:
Hey @nanory,
ok, just check my latest commit.